ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,866 @@
+    my $status = $self->execute_command($command, $why, $keeps_state, $stdin);
+
+    return $status if $NoAction;
+
+    my @lines = split /\n/, $cmd_all;
+    my $packages_to_remove = Set::Scalar->new();
+    foreach my $line (@lines) {
+	my $cf = index($line, 'conflicts with file from package');
+	if ( $cf != -1) {	# handle file conflicts
+	    my $pkg = $line;
+	    $pkg =~ s,^.*package ,,;
+	    $pkg =~ s,-[0-9]*:,-,;
+	    $self->warn("$pkg will be removed to avoid file conflict");
+	    $packages_to_remove->insert($pkg) if !$packages_to_remove->has($pkg);
+	} else {
+	    $cf = index($line, 'Error: ');
+	    my $pkg = $line;
+	    if ( $cf != -1 && index($line, 'conflicts with') ) {	# handle package conflicts
+		if ( $os_major eq "5" ) {
+		    $pkg =~ s,^.*Error: ,,;
+		} else {
+		    $pkg =~ s,^.*Error: Package: ,,;
+		}
+		$pkg =~ s, .*,,;
+		$self->warn("$pkg will be removed to avoid package conflict");
+		$packages_to_remove->insert($pkg) if !$packages_to_remove->has($pkg);
+	    } else {
+		my $dep = index($line, 'is needed by (installed)');
+		if ( $dep != -1) {
+		    my $pkg = $line;
+		    $pkg =~ s,^.*\(installed\) ,,;
+		    $pkg =~ s,-[0-9]*:,-,;
+		    $self->warn("$pkg will be removed to avoid file conflict");
+		    $packages_to_remove->insert($pkg) if !$packages_to_remove->has($pkg);
+		}
+	    }
+	}
+    }
+
+    if ( $packages_to_remove->size ) {
+	$status = $self->execute_command([ "yum remove -y " . YUM_PLUGIN_OPTS . " " . join (" ", sort @$packages_to_remove) ], 'attempting to remove: '.join (" ", sort @$packages_to_remove), $keeps_state, $stdin);
+
+	if ( index($cmd_all, 'Error: ') != -1 ) {
+	    $self->error("Unable to remove conflicting packages.");
+	} else {
+	    $status = $self->execute_command($command, 'retrying: '.$why, $keeps_state, $stdin);
+	}
+    }
+
+    return $status;
+}
+
+sub Configure
+{
+    my ( $self, $config ) = @_;
+
+    # Make sure stdout and stderr is flushed every time to not to
+    # put mess on serial console
+    autoflush STDOUT 1;
+    autoflush STDERR 1;
+
+    # We are parsing some outputs in this component. We must set a
+    # locale that we can understand.
+    local $ENV{LANG}   = 'C';
+    local $ENV{LC_ALL} = 'C';
+
+    my $repos = $config->getElement(REPOS_TREE)->getTree();
+    my $t     = $config->getElement(CMP_TREE)->getTree();
+    
+    # Display system info
+    if ( defined($t->{quattor_os_release}) ) {
+	$self->info("target OS build: ", $t->{quattor_os_release});
+    }
+    
+    # Detect OS
+    my $fhi;
+    if ( open( $fhi, '<', "/etc/redhat-release" ) ) {
+        while ( my $line = <$fhi> ) {
+	    my $i = index($line, 'release ');
+            if ( $i >= 0 ) {
+		chomp($line);
+		$self->info("local OS: ".$line);
+		$os_major = substr($line, $i+8, 1);
+                last;
+            }
+        }
+        $fhi->close();
+    }
+
+    if ( $os_major eq "" ) {
+        $self->error("Unable to determine OS release.");
+	return 0;
```

Comment:
Really?! Surely 1 should only be returned if it's successful? Indentation has been fixed in our internal repo, will be in next push.